### PR TITLE
Extract name

### LINKED ISSUE #242: Possible naming of JSON file produced by `extract`
This could help users to know their JSON files.
Moreover this would be necessary for users to perform the tests with pytest as described in the documentation.

### PR DESCRIPTION
**Pull request by 14thibea**
Closes #242.
When the user choses the name of the JSON file, it may not follow the convention `extract_XXX.json` anymore.
If the user forgets to add the JSON extension on writing or reading, it is added automatically by ClinicaDL.

### CODE REVIEW

**Comment by ravih18** [comment]
Hi, 

I just have a comment about the name of the option / variable `json_name`. I think it is better to keep the convention we use with the extension at the end of the option name: `xxxxx_json` instead of `json_name`. It could be `extract_json`, `preprocessing_json` (like in train command) or `output_json` for instance.

Otherwise it looks good to me.

File: clinicadl/extract/extract_utils.py [hunk 1]
```diff
@@ -43,9 +46,20 @@ def get_parameters_dict(
         parameters["acq_label"] = acq_label
         parameters["suvr_reference_region"] = suvr_reference_region
 
+    parameters["json_name"] = compute_json_name(json_name)
+
     return parameters
 
 
+def compute_json_name(json_name: str) -> str:
+    if json_name is None:
```
Comment (ravih18) [inline clinicadl/extract/extract_utils.py]: I am not sure, but didn't we decide to change our None test by `if not ..` instead of `if .. is None` ?

Comment (omar-rifai) [reply clinicadl/extract/extract_utils.py]: Just for the records, we had to backtrack on some of these changes in Clinica. Particularly for empty pandas' Dataframes which crash on the check `if not <dataframe>`. For `str` it is indeed more idiomatic.

Comment (14thibea) [reply clinicadl/extract/extract_utils.py]: We changed it in the argument detection, but not because of the style: it allowed to include in the test empty lists and tuples (with None values).
I didn't think it was a convention to follow everywhere

**Comment by 14thibea** [comment]
Thank you for the review :)
I renamed the variable `extract_json` it's better!
And I added it in the documentation also.

**Review by mdiazmel — APPROVED** [review]
Looks good. Ready to merge!
